src/dataflow/src/source/kafka.rs

```diff
@@ -724,23 +668,28 @@ where
             worker_id,
             worker_count,
         );
-        dp_info.refresh_source_information(&mut cp_info, &activator);
 
         move |cap, output| {
+            let timer = Instant::now();
+
+            dp_info.source_metrics.operator_scheduled_counter.inc();
+
             // Accumulate updates to BYTES_READ_COUNTER;
             let mut bytes_read = 0;
-            dp_info.source_metrics.operator_scheduled_counter.inc();
-            // Repeatedly interrogate Kafka for messages. Cease when
-            // Kafka stops returning new data, or after 10 milliseconds.
-            let timer = std::time::Instant::now();
+
+            // Trigger any waiting librdkafka callbacks. This should never
+            // return any messages, because we've set things up to receive those
+            // via individual partition queues.
+            {
+                let message = dp_info.consumer.poll(Duration::from_secs(0));
```

Comment:
I think there's a risk that if we call poll first (before calling split_partition_queue) we'll get a message there.

Reply:
Yep, my bad for the consuming docs. I put an explanation of why this ok above.